### PR TITLE
Hiding dashboard cards: make quick start cards personalizable

### PR DESCRIPTION
**Pull request by kean**
Related issue: #20296

Previous PRs (dependencies):
1. #20365
2. #20369
3. #20384

> I can't set the previous PR #20385 as a target because it only exists in my fork. If you want to review only the changes from this PR, see the following [diff](https://github.com/wordpress-mobile/WordPress-iOS/pull/20393/files/9d2cd357e32f18eaa52b693a054d4aded460d4a5..672ee148ba223d8d0eb2296f5b869b1df1bceec0).

Add personalization support for "Next steps" and "Get to know the app" cards. Previously, tapping "Remove next steps" on the Home tab worked by removing all active tours and setting `quickStartType` to `.undefined`. And now it's just a matter of setting a flag in the personalization service introduced in #20365. It allows the user to continue the tour or revisit a completed one.

## Changes:

- Add "Next steps" and "Get to know the app" cards to the Personalize Home Tab screen.
- Update the design of the context menu for the quick start card (use a context menu instead of an action sheet)
- Add unit test to make sure all personalizable cards have a predefined UserDefaults key and titles for menus to catch regressions in case someone make an existing card personalizable without updating the other places

## Testing

*Prerequisites:*

- Feature Flag "Personalize Home Tab" enabled

**Test 1**

- Install the app (fresh install)
- Log in and select "Not sure, show me around" on the "What would you like to focus on first" page
- Complete one of the Quick Start steps
- On the Home tab, open the context menu for the "Get to know the app" card
- Verify that the context menu uses the new design (context menu instead of an action sheet)
- Tap "Hide this"
- Verify that the "Get to know the app" card was removed from the Home tab
- Tap "Personalize Home Tab" at the bottom of the Home tab
- Verify that the "Get to know the app" switch is displayed and it's disabled
- Toggle the switch and close the screen
- Verify that the "Get to know the app" is again displayed on the Home tab 
- Verify that the Quick Start is restarted from the previous state
- Try going through one of the next steps and verify that it works

**Test 2**

- Repeat test 1, but instead of completing only one step, complete all of them
- Verify that you can still show/hide the "Get to know the app" card at any time to revisit the completed steps

**Test 3**

- Repeat tests 1 and 2 for the "Next steps" card. An easy way to enable it is by using the "Enable Quick Start for New Site" button from the Debug menu.

**Test 4**

- Install the app (fresh install)
- Login and tap "Skip" on the "What would you like to focus on first" page to make sure no Quick Start guides are added
- Tap "Personalize Home Tab" at the bottom of the Home tab
- Verify that neither the "Next steps" nor "Get to know the app" cards are displayed in the list

## Screenshots

Screenshot 2023-03-23 at 7 56 16 PM Screenshot 2023-03-23 at 7 56 31 PM

## Regression Notes

1. Potential unintended areas of impact: 
Anything that's related to the quick start guides ("Next steps" or "What would you like to focus on first")
4. What I did to test those areas of impact (or what existing automated tests I relied on):
I performed manual testing and verified that the guides still work
5. What automated tests I added (or what prevented me from doing so)
I covered new changes with unit tests (making sure that the switch is displayed in the "Personalize Home Tab" screen)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/BlogDashboardPersonalizeCardCell.swift
Comment (kean) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/BlogDashboardPersonalizeCardCell.swift:81]: I considered injecting the entire blog, but it would've made it harder to unit-test and create previews for this screen. I don't expect more parameters to be needed.

Comment (staskus) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/BlogDashboardPersonalizeCardCell.swift:81]: Got it! Thanks for including the diff for an easier review. 

The injection of `quickStartType` was the only question mark for me. However, although at first sight it did catch my attention, I think this approach is more correct even beyond the argument of making it easier to write unit-tests and create previews. We make `BlogDashboardPersonalizationViewModel` depend on only what it needs to depend on. Due to business requirements, a quick start does require special handling. 

>  I don't expect more parameters to be needed.

Yes, it doesn't look to be needed for the existing cards. In the future, who knows. If more and more custom behaviour would be needed, we could consider some different approach which wouldn't require injecting all the different dependencies into personalization VM and maybe would be handled by some configuration of each card.

**Review by staskus — COMMENTED** [review]
👍 The functionality works well. 

I reviewed the code as well, I get the injection of `quick start`.

Since this PR comes with the changes of many different PRs, I will approve it once it only contains the changes it needs. 

@kean I know it's inconvenient, so just ping me once that happens.

**Comment by kean** [comment]
Hey, @staskus. I rebased it and added a [convenience API](https://github.com/wordpress-mobile/WordPress-iOS/pull/20393/commits/013abb90dfd4624dde052e2764a3eb4640f34eea) for setting the card menu, as you suggested on Friday.

**Comment by staskus** [comment]
> Hey, @staskus. I rebased it and added a [convenience API](https://github.com/wordpress-mobile/WordPress-iOS/pull/20393/commits/013abb90dfd4624dde052e2764a3eb4640f34eea) for setting the card menu, as you suggested on Friday.

@kean thanks!

**Review by staskus — APPROVED** [review]
👍 Thank you for the changes, looks good. 

I ran CI.

cc @guarani if you sill want to take a look. Otherwise, I'll merge it.

**Comment by kean** [comment]
Hey, @staskus  / @Guarani. I rebased this PR and ran a quick test to ensure it still works.

**Comment by staskus** [comment]
@kean  Thanks! Looks good to me 👍. Let's wait for CI builds to finish running and merge this change. 

 cc: @guarani, @momo-ozawa

**Comment by momo-ozawa** [comment]
Thanks for your work @kean! Let's target the 22.6 milestone. I've created a separate issue for updating release notes https://github.com/wordpress-mobile/WordPress-iOS/issues/20775

**Comment by guarani** [comment]
Nice to see this land! I'll skip testing since it looks like others have already, but let me know if you'd like me to.

**Pull request merged**
